Release117

orhankislal opened the pull request:
<!--  

Thanks for sending a pull request!  Here are some tips for you:
1. Refer to this link for contribution guidelines https://cwiki.apache.org/confluence/display/MADLIB/Contribution+Guidelines
2. Please Provide the Module Name, a JIRA Number and a short description about your changes.
-->

- [x] Add the module name, JIRA# to PR/commit and description.
- [x] Add tests for the change.

orhankislal commented:
jenkins retest this

fmcquillan99 commented on RELEASE_NOTES:
```diff
@@ -12,10 +12,66 @@ Current list of bugs and issues can be found at https://issues.apache.org/jira/b
+    - MADlib cannot be built on PowerPC machines with Linux (MADLIB-1410)
+    - Pivot:  Pivot documentation should say "out_table" instead of "output_table" (MADLIB-1376)
 
 Other:
```
Please add to this section 
```
- DL: CUDA GPU memory cannot be released until the process holding it is terminated.  This process holds the GPU memory until one of the following two things happen: query finishes and user logs out of the Postgres client/session; or, query finishes and user waits for the timeout set by `gp_vmem_idle_resource_timeout`. The default value for this timeout in Greenplum is 18 sec, but it can be changed.
```

orhankislal replied:
Should this go in `big fixes` or `other`?

fmcquillan99 replied:
other

reductionista commented on src/ports/postgres/modules/deep_learning/madlib_keras_fit_multiple_model.sql_in:
```diff
@@ -1016,7 +1016,7 @@ name                      |
 start_training_time       | 2019-12-16 18:54:33.826414
 end_training_time         | 2019-12-16 18:56:19.106321
-madlib_version            | 1.17-dev
+madlib_version            | 1.17
```
Maybe these should also say 1.17.0?

Also:  it doesn't show up here in the PR, but in `deep_learning/madlib_keras_fit.sql_in` the versions in the examples still say 1.16

orhankislal commented:
> I tried the upgrade on my MacBookPro, from 1.16 to 1.17.0 and it threw an exception:
> 
> ```
> ERROR:  column p.prokind does not exist
> LINE 12:                         p.prokind = 'a' AS proisagg
>                                  ^
> HINT:  Perhaps you meant to reference the column "p.probin".
> : ERROR : False
> ```
> 
> This is with postgres 10.5:
> PostgreSQL 10.5 on x86_64-apple-darwin17.7.0, compiled by Apple LLVM version 10.0.0 (clang-1000.11.45.2), 64-bit

The prokind column does not exist in PG10 (it has `proisagg` and `proiswindow` for the same purposes). Since we only support 2 versions of postgres, we dropped PG10 and added PG12 with this release. Compiling etc. should still work for PG10 but the upgrade process does use this particular column.

reductionista commented:
Upgrade succeeded on my MacBookPro with gpdb5
```
PostgreSQL 8.3.23 (Greenplum Database 5.23.0+dev.10.g50015fed64 build dev) on x86_64-apple-darwin18.7.0, compiled by GCC Apple LLVM version 10.0.1 (clang-1001.0.46.4), 64-bit compiled on Nov  5 2019 08:49:26 (with assert checking)
```
LGTM!
